apps/browser/lib/services/evo/EvoThread.ts

```diff
@@ -96,23 +97,26 @@ export class EvoThread {
     this._callbacks.setWorkspace(this._state.workspace);
   }
 
-  async start(options: EvoThreadStartOptions) {
+  async start(options: EvoThreadStartOptions): Promise<void> {
     const {
       goal,
       allowTelemetry,
       openAiApiKey
     } = options;
 
-    // Wait until loading has finished
-    await this.waitForLoad();
-
     if (this._state.isRunning) {
-      this._callbacks?.onError("A goal is already underway.");
+      if (this._state.goal !== options.goal) {
+        this._callbacks?.onError("A goal is already underway.");
+      }
```

Comment:
is this needed? if an evo thread is already running and the user tries to execute the `start` method shouldn't we return the error?

Reply:
Yes because we cannot guarantee "handleStart" won't be called twice due to react invoking useEffect callbacks multiple times in dev. This was an annoying bug, and this fixes it by moving state outside of the component into the evo thread.